stl/inc/limits

```diff
@@ -1050,11 +1050,8 @@ _NODISCARD constexpr int _Popcount_fallback(_Ty _Val) noexcept {
     _Val = static_cast<_Ty>((_Val & static_cast<_Ty>(0x3333'3333'3333'3333ull))
                             + ((_Val >> 2) & static_cast<_Ty>(0x3333'3333'3333'3333ull)));
     _Val = static_cast<_Ty>((_Val + (_Val >> 4)) & static_cast<_Ty>(0x0F0F'0F0F'0F0F'0F0Full));
-    for (int _Shift_digits = 8; _Shift_digits < _Digits; _Shift_digits <<= 1) {
-        _Val = static_cast<_Ty>(_Val + static_cast<_Ty>(_Val >> _Shift_digits));
-    }
-    // we want the bottom "slot" that's big enough to store _Digits
-    return static_cast<int>(_Val & static_cast<_Ty>(_Digits + _Digits - 1));
+    _Val = static_cast<_Ty>(_Val * static_cast<_Ty>(0x0101'0101'0101'0101ull));
+    return static_cast<int>(_Val >> (_Digits - 8));

```

Comment:
This is not obvious
```suggestion
   // Multiply by one in each byte, so that it will have the sum of all source bytes in the highest byte
    _Val = static_cast<_Ty>(_Val * static_cast<_Ty>(0x0101'0101'0101'0101ull));
   // Extract highest byte
    return static_cast<int>(_Val >> (_Digits - 8));
```

Reply:
> I don't understand what is 0x0101'0101'0101'0101 and why we multiply it.

Oh, I see this is not obvious to you too :-)

Recall long multiplication from school;
```
        a b c d 
     x  1 1 1 1
       --------
        a b c d
+     a b c d
    a b c d
  a b c d
  -------
  * * * R # # #
```
Since `a`, `b`, `c`, and `d` are either 0 or 1, and digit can be up to 256, there's no carry during sum.
So R = a+b+c+d. `*` are reomved due to cast overflow, and `#` are removed by the last shift